services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java

```diff
@@ -173,12 +174,15 @@ public ValueSets constructProtobufMapping() throws UnableToLoadFileException {
    */
   private ValueSet read(String valueSetId, ConfigValueProvider configGetter, String valueSetDefaultPath)
       throws UnableToLoadFileException {
-    Optional<ValueSet> result = getValueSet(valueSetId).or(() -> {
+    Optional<ValueSet> result = getValueSet(valueSetId);
+
+    if (result.isEmpty()) {
       String path = configGetter.getPath(distributionServiceConfig.getDigitalGreenCertificate());
       return readConfiguredJsonOrDefault(resourceLoader, path, valueSetDefaultPath,
           ValueSet.class);
-    });
-    return result.orElseThrow(() -> new UnableToLoadFileException(valueSetDefaultPath));
+    }
+
+    return result.get();
```

Comment:
Why not throwing a custom exception here (UnableToLoadFileException or DigitalCovidCertificateException)? We don't want the NoSuchElementException here!